examples/initCPGrid.hpp

```diff
@@ -0,0 +1,47 @@
+/*
+  This is small utility to inclded in the example/test programs in opm-upscaling.
+*/
+
+#ifndef INIT_CPGRID_HPP
+#define INIT_CPGRID_HPP
+
+#include <array>
+
+#include <opm/parser/eclipse/Parser/Parser.hpp>
+#include <opm/parser/eclipse/Parser/ParseContext.hpp>
+#include <opm/parser/eclipse/Deck/Deck.hpp>
+#include <opm/parser/eclipse/EclipseState/EclipseState.hpp>
+
+/// Initialize the grid.
+void initCPGrid(Dune::CpGrid& grid, const Opm::parameter::ParameterGroup& param)
+{
+    std::string fileformat = param.get<std::string>("fileformat");
+    if (fileformat == "sintef_legacy") {
+        std::string grid_prefix = param.get<std::string>("grid_prefix");
+        grid.readSintefLegacyFormat(grid_prefix);
+    } else if (fileformat == "eclipse") {
+        std::string filename = param.get<std::string>("filename");
+        if (param.has("z_tolerance")) {
+            std::cerr << "****** Warning: z_tolerance parameter is obsolete, use PINCH in deck input instead\n";
+        }
+        bool periodic_extension = param.getDefault<bool>("periodic_extension", false);
+        bool turn_normals = param.getDefault<bool>("turn_normals", false);
+        Opm::ParseContext parseContext;
+        Opm::ParserPtr parser(new Opm::Parser());
+        Opm::DeckConstPtr deck(parser->parseFile(filename , parseContext));
+        std::shared_ptr<Opm::EclipseState> state = std::make_shared<Opm::EclipseState>(deck , parseContext);
+        grid.processEclipseFormat(state->getInputGrid(), periodic_extension , turn_normals );
+    } else if (fileformat == "cartesian") {
+        std::array<int, 3> dims = {{ param.getDefault<int>("nx", 1),
+                                     param.getDefault<int>("ny", 1),
+                                     param.getDefault<int>("nz", 1) }};
+        std::array<double, 3> cellsz = {{ param.getDefault<double>("dx", 1.0),
+                                          param.getDefault<double>("dy", 1.0),
+                                          param.getDefault<double>("dz", 1.0) }};
+        grid.createCartesian(dims, cellsz);
+    } else {
+        OPM_THROW(std::runtime_error, "Unknown file format string: " << fileformat);
+    }
+}
+
+#endif
```

Comment:
> [...] stick it in opm/upscaling 

As in extend the upscaling library? The reason I did it the way I did was that (I assumed) the example programs could only consist of one compilation unit. Happy to add header and source files in `opm/upscaling`.